calico/common.py

```diff
@@ -622,6 +703,29 @@ def validate_tags(profile_id, tags):
         raise ValidationFailed(" ".join(issues))
 
 
+def validate_labels(profile_id, labels):
+    issues = []
+
+    if not VALID_ID_RE.match(profile_id):
+        issues.append("Invalid profile_id '%r'." % profile_id)
```

Comment:
This is a different piece of the datamodel `calico/v1/policy/profile/<id>/labels` vs `calico/v1/policy/profile/<id>/rules`